apps/ca.c

```diff
@@ -985,36 +979,34 @@ int ca_main(int argc, char **argv)
 
         if (verbose)
             BIO_printf(bio_err, "writing new certificates\n");
-        for (i = 0; i < sk_X509_num(cert_sk); i++) {
-            BIO *Cout = NULL;
-            ASN1_INTEGER *serialNumber = X509_get_serialNumber(x);
-            int k;
-            char *n;
-
-            x = sk_X509_value(cert_sk, i);
 
-            j = ASN1_STRING_length(serialNumber);
-            p = (const char *)ASN1_STRING_get0_data(serialNumber);
+        new_cert[PATH_MAX - 1] = '\0';
```

Comment:
There's really no need for this at all, since the following `strcpy` will place the NUL character in its correct position anyway.